tests/mocks.py

```diff
@@ -117,6 +117,32 @@ def prepare(self):
         return super(EngineEmul, self).prepare()
 
 
+# Allow custom configs
+class EngineEmul2(Engine):
```

Comment:
I see no reason to have two test Engines. In my point for results consistency we must have single one only which behavior corresponding to real Engine. Do you agree?

Reply:
Yes, those codes are ugly. 
I merged the changes into the original class now.